packages/grid/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -708,4 +709,9 @@ export interface DataGridPropsWithoutDefaultValue extends CommonProps {
     newRow: GridRowModel,
     oldRow: GridRowModel,
   ) => Promise<GridRowModel> | GridRowModel;
+  /**
+   * If `true`, the reordering of rows is enabled.
+   * @default false
+   */
+  enableRowReorder?: boolean;
```

Comment:
This in a pro-only prop with a default value

1. Move this line to `DataGridProPropsWithDefaultValues` without the `?`
2. Add a default value to `DATA_GRID_PRO_PROPS_DEFAULT_VALUES`
3. Remove the forced prop in `enableRowReorder` and `DATA_GRID_FORCED_PROPS`


That way you have `?: boolean` for `DataGridProProps` and `: boolean` for `DataGridProProcessedProps` which is used internally.
And on the community plan, the props is never used so if the user passes `true` it won't matter (like for the tree data for instance).

____

Concerning the name
We use `props.pagination` not `props.enablePagination`, same for `props.treeData`
Should we use `props.rowReordering` for instance ?